crates/uv-resolver/src/resolution.rs

```diff
@@ -316,6 +319,141 @@ impl ResolutionGraph {
     pub fn petgraph(&self) -> &petgraph::graph::Graph<Dist, Range<Version>, petgraph::Directed> {
         &self.petgraph
     }
+
+    /// Return the marker tree specific to this resolution.
+    ///
+    /// This accepts a manifest, in-memory-index and marker environment. All
+    /// of which should be the same values given to the resolver that produced
+    /// this graph.
+    ///
+    /// The marker tree returned corresponds to an expression that, when true,
+    /// this resolution is guaranteed to be correct. Note though that it's
+    /// possible for resolution to be correct even if the returned marker
+    /// expression is false.
+    ///
+    /// For example, if the root package has a dependency `foo; sys_platform ==
+    /// "macos"` and resolution was performed on Linux, then the marker tree
+    /// returned will contain a `sys_platform == "linux"` expression. This
```

Comment:
So what we do is collect all marker keys that are used and add them with the current platform, which is stricter than necessary (unlike add `sys_platform != "macos"`)?

Reply:
That's correct. This is necessary since the proposal only allows you to provide a dictionary of required markers, so we can't express `sys_platform != "macos"` -- only `sys_platform == "whatever_we_have"`.